test/typescript-tests/testTypes.ts

```diff
@@ -2308,36 +2299,3 @@ Random examples
     MathJsChain<number[]>
   >()
 }
-
-/*
-MathNode examples
-*/
-{
-  class CustomNode extends Node {
-    a: MathNode
-    constructor(a: MathNode) {
-      super()
-      this.a = a
-    }
-  }
-
-  // Basic node
-  const instance1 = new Node()
-
-  // Built-in subclass of Node
-  const instance2 = new ConstantNode(2)
-
-  // Custom subclass of node
-  const instance3 = new CustomNode(new ConstantNode(2))
-
-  expectTypeOf(instance1).toMatchTypeOf<MathNode>()
-  expectTypeOf(instance1).toMatchTypeOf<MathNodeCommon>()
```

Comment:
If we're trying for a discriminated union on MathNode, then I don't think this test *should* pass: `instance1` is a `BaseNode`, and I don't think a BaseNode should be assignable to a MathNode: BaseNodes. The `type` property on BaseNode is `type: string`, whereas on all other nodes it's a constant.

If `BaseNode` is assignmable to `MathNode`, then
```ts
let node: MathNode
if (node.type === 'AssignmentNode) {
    // now node has type BaseNode | AssignmentNode
}
```